x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -128,7 +129,7 @@ void handleResponse(InternalInferModelAction.Response response,
         Map<String, Object> fields = new HashMap<>(ingestDocument.getSourceAndMetadata());
         if (fieldMapping != null) {
             fieldMapping.forEach((src, dest) -> {
-                Object srcValue = fields.remove(src);
```

Comment:
@davidkyle I suppose. This would require the `dig` method to have a sister "remove" method. For simplicity I opted not to include it. Having a larger map passed around is pretty cheap right now. We may have issues ones we support more than a ~100 features.